### ISSUE TITLE
 Getting EGL_BAD_NATIVE_WINDOW and EGL_BAD_DISPLAY errors when using MediaPipe on Android

### ISSUE THREAD

**Issue by V-m1r** [google/mediapipe#4617]
Platform & Solution: Pose-tracking Android AAR
Mediapipe version 0.8.9 (Can't update mediapipe to a newer version (writing another issue))

Hello,
I am currently working on an Android application that uses the MediaPipe pose-tracking aar for physical exercise recognition. The application works by capturing video, processing it through MediaPipe, and displaying the output.

However, I've been encountering some issues when attempting to manage the lifecycle of the SurfaceView used for the output display. I am trying to resize the previewdisplayview at runtime (only once) and after an attempt to resize the surfaceview I am getting EGL_BAD_NATIVE_WINDOW and EGL_BAD_DISPLAY errors, which cause the app to crash. THIS HAPPENS OCCASIONALLY (3/10 times the app will crash). Here is the method that I call: 
```
    private void reduceFrame() {

        ConstraintLayout.LayoutParams layoutParams = (ConstraintLayout.LayoutParams) b.cvMediapipe.getLayoutParams();
        layoutParams.height = dpToPx(260);
        layoutParams.width = dpToPx(134);
        layoutParams.setMargins(0, dpToPx(10), dpToPx(10), 0);
// so mediapipe view is initially fullscreen and after a period of time I am trying to resize the layouts to position mediapipe frame layout in the right top corner of the screen 
        runOnUiThread(() -> {
// cvMediapipe is a CardView to give Mediapipe view a rounded corner, so FrameLayout is inside the CardView (see .xml below)
            b.cvMediapipe.setLayoutParams(layoutParams);
            processor.getVideoSurfaceOutput().setSurface(null);
            b.viewGroup.removeView(previewDisplayView);

            previewDisplayView = new SurfaceView(this);
            b.viewGroup.addView(previewDisplayView);
        });


    }
```
My .xml: 
```
    <androidx.constraintlayout.widget.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
... > 

    <androidx.cardview.widget.CardView
            android:id="@+id/cv_mediapipe"
            android:layout_width="match_parent"
            android:layout_height="match_parent"

            app:cardBackgroundColor="@color/transparent"
            app:cardCornerRadius="40dp"
            app:cardPreventCornerOverlap="false"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintTop_toTopOf="parent">

            <FrameLayout
                android:id="@+id/view_group"
                android:layout_width="match_parent"
                android:layout_height="match_parent" />

        </androidx.cardview.widget.CardView>

```

Here's a snippet of the error log:

```
2023-07-17 02:47:14.334 31043-31043/myapplication.osnownoy_fit E/libc: Access denied finding property "ro.vendor.df.effect.conflict"
2023-07-17 02:47:14.334 31043-31043/myapplication.osnownoy_fit E/libc: Access denied finding property "ro.vendor.knock.type"
2023-07-17 02:47:21.598 31043-31043/myapplication.osnownoy_fit E/libEGL: validate_display:87 error 3008 (EGL_BAD_DISPLAY)
2023-07-17 02:47:21.791 31043-31246/myapplication.osnownoy_fit E/libc: Access denied finding property "persist.vendor.camera.privapp.list"
2023-07-17 02:47:25.036 31043-31230/myapplication.osnownoy_fit E/libEGL: eglCreateWindowSurfaceTmpl:708 error 300b (EGL_BAD_NATIVE_WINDOW)
2023-07-17 02:47:25.037 31043-31293/myapplication.osnownoy_fit A/libc: Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 31293 (Thread-15), pid 31043 (on.osnownoy_fit)
```
The issue occurs when I try to resize the SurfaceView in reduceFrame method.

I've tried several solutions, such as ensuring all operations are performed on the main UI thread, updating the SurfaceView's size rather than recreating it, waiting until the SurfaceHolder.Callback is called before setting the surface to the MediaPipe processor, and more. However, none of these have resolved the issue.

I've tested the application on multiple devices, but the issue persists.

Does anyone have any ideas on what could be causing these errors and how to resolve them? Any help would be greatly appreciated!

Thanks

**Comment by V-m1r** [comment]
Fixed it. The issue was in incorrect usage and management of SurfaceView. 
Setting  processor.getVideoSurfaceOutput().setSurface(null); is a mistake when the frames are still being fed to the processor. A way to handle it is to wait until surfaceView gets changed (through callback listener) and then setSurface on processor
